src/plugins/noAutoUnmute/index.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Vencord, a modification for Discord's desktop app
+ * Copyright (c) 2024 Vendicated and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+
+export default definePlugin({
+    name: "NoAutoUnmute",
+    description: "Prevents you from being automatically unmuted by Discord (such as by joining a VC call)",
+    authors: [Devs.HAHALOSAH],
+    patches: [
+        {
+            find: "isSelfMute(){",
+            replacement: {
+                match: /video:\i}=\i;/,
+                replace: "$&return;"
```

Comment:
the function you're fully nooping here is also responsible for other stuff, eg the no input detected notice
![image](https://github.com/Vendicated/Vencord/assets/45497981/919a9a42-b507-457d-aa49-e8fe657c7c30)
![image](https://github.com/Vendicated/Vencord/assets/45497981/bcae9062-306f-4cef-b5d9-e074ee5802b4)

so a more precise patch would be better

```suggestion
                match: /\(\i.mute||\i.deaf\)&&/,
                replace: "false&&"
```